src/lib/Parallax/ParallaxContext.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { createContext, PropsWithChildren, useContext, useEffect, useState, useRef } from 'react';
+
+interface ParallaxContextProps {
+  yOffsetRatio: number;
+}
+
+const ParallaxContext = createContext<ParallaxContextProps>({ yOffsetRatio: 0 });
+
+export const useScrollContext = () => useContext(ParallaxContext);
+
+interface ParallaxProviderProps {
+  css: React.CSSProperties;
+}
+
+export const ParallaxProvider = ({ css, children }: PropsWithChildren<ParallaxProviderProps>) => {
+  const parallaxRef = useRef<HTMLDivElement | null>(null);
+  const [yOffsetRatio, setYOffsetRatio] = useState(0);
+
+  useEffect(() => {
+    const handleScroll = () => {
+      if (parallaxRef.current) {
+        const windowY = window.scrollY;
+        const elementOffsetTop = parallaxRef.current.offsetTop;
+
+        const centerOffset = windowY - elementOffsetTop;
+        const elementHeight = parallaxRef.current.offsetHeight;
+
+        const offsetRatio = centerOffset / elementHeight;
+
+        setYOffsetRatio(offsetRatio);
+      }
+    };
+
+    const observer = new IntersectionObserver(([entry]) => {
+      if (entry.isIntersecting) {
+        window.addEventListener('scroll', handleScroll, { passive: true });
```

Comment:
passive 속성을 처음봐서 찾아보니 스크롤 성능과 관련되어있다고 봤습니다.

이 속성을 false 를 지정해도 육안으로 확인되는 차이는 없었던거 같은데 어떤 이점이 있을까요?? 단순 궁금함입니다. 😃

Reply:
맞습니다 실제로 해당 옵션을 true로 해도 워낙 요즘 브라우저는 최적화가 잘 되어있어서 성능 이점을 느끼기 어렵다고 하더라고요. 그래도 학습한 김에 달아보았습니다!

라고 달다가 다시 한번 찾아봤는데... 뒤늦게 알아버렸지만, 기본 Scroll 이벤트에 대해서는 취소가 불가능하기 때문에, 리스너가 페이지 랜더링을 블락할 수 없다고 합니다...하하.. 역시 문서가 중요하네요..
[MDN](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#using_passive_listeners)
>You don't need to worry about the value of passive for the basic [scroll](https://developer.mozilla.org/en-US/docs/Web/API/Element/scroll_event) event. Since it can't be canceled, event listeners can't block page rendering anyway.

Reply:
오우 이런 부분이 있었군요.. 레퍼런스 감사합니다 👍👍